doc/build/sysbuild/index.rst

```diff
@@ -619,6 +619,51 @@ configuration files for ``my_sample`` will be ignored.
 This give you full control on how images are configured when integrating those
 with ``application``.
 
+.. _sysbuild_zephyr_application_dependencies:
+
+Adding dependencies among Zephyr applications
+=============================================
+
+Sometimes, in a multi-image build, you may want certain Zephyr applications to
+be configured or flashed in a specific order. For example, if you need some
+information from one application's build system to be available to another's,
+then the first thing to do is to add a configuration dependency between them.
+Separately, you can also add flashing dependencies to control the sequence of
+images used by ``west flash``; this could be used if a specific flashing order
+is required by an SoC, a _runner_, or something else.
+
+By default, sysbuild will configure and flash applications in the order that
+they are added, as ``ExternalZephyrProject_Add()`` calls are processed by CMake,
+but this order is not well defined. To make adjustments according to your needs,
+you can use the ``sysbuild_add_dependencies()`` function. Its usage is similar
+to the standard ``add_dependencies()`` function in CMake.
+
+Here is an example of adding configuration dependencies for ``my_sample``:
+
+.. code-block:: cmake
+
+   sysbuild_add_dependencies(IMAGE CONFIGURE my_sample sample_a sample_b)
+
+This will ensure that sysbuild will run CMake for ``sample_a`` and ``sample_b``
+(in some order) before doing the same for ``my_sample``, when building these
+domains in a single invocation.
+
+If you want to add flashing dependencies instead, then do it like this:
+
+.. code-block:: cmake
+
+   sysbuild_add_dependencies(IMAGE FLASH my_sample sample_a sample_b)
+
+As a result, ``my_sample`` will be flashed after ``sample_a`` and ``sample_b``
+(in some order), when flashing these domains in a single invocation.
+
+.. tip::
+
+   Unlike with ``add_dependencies()``, you are allowed to use
+   ``sysbuild_add_dependencies()`` with non-existent images. This makes it
+   possible to define relationships between various images in the build system,
+   regardless of whether they are present in the current multi-image build.
```

Comment:
Just saw this, I do not like this idea, why would you want to sort on images that are not present? They shouldn't be included in a build.

Reply:
I intended this initially to simplify adding dependencies for a conditionally included image, thus avoiding a lot of this:
```
if mcuboot exists:
  mcuboot depends on my_sample
endif
```

Similarly for groups:
```
if mcuboot exists:
  add mcuboot to my_group
endif
```

But on second thought, maybe these cases won't even be as common as I first imagined. I'll consider raising errors, just like the standard `add_dependencies()`.

Reply:
Actually, I just remembered that since the processing order of CMake files is not well defined, there might not even be a good way to check `if <image> exists`.

Reply:
After all images have been configured should be OK.

Reply:
At this point, I expect all groups and dependencies to be added as well. They are resolved before the PRE_CMAKE hooks.

Reply:
> Those will be processed in the order they are reported by `west list` (or `ZEPHYR_MODULES` when `west` is not used)

That's neat! Based on this fact, I added another change to make the order of including image-specific `sysbuild.cmake` well-defined. With that, I can comfortably address the lack of errors from `sysbuild_add_dependencies()`.